extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -14,6 +19,58 @@ export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private @const {?../../amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {?../../../src/service/localization.LocalizationService} */
+    this.localizationService_ = null;
+  }
+
+  /**
+   * Updates the CTA based on the shopping data.
+   * @param {string} pageIndex
+   * @param {string} pageId
+   * @private
+   */
+  updateCta_(pageIndex, pageId) {
+    const shoppingTag = this.element.ownerDocument
+      .getElementById(pageId)
+      .getElementsByTagName('amp-story-shopping-tag');
+    const numShoppingTags = shoppingTag != null ? shoppingTag.length : 0;
+
+    const pageOutlink = document.getElementsByClassName(
+      'i-amphtml-story-page-open-attachment-host'
+    )[pageIndex].shadowRoot.children[1];
+
+    if (numShoppingTags === 1) {
+      this.mutateElement(() => {
+        const productName = this.storeService_.get(StateProperty.SHOPPING_DATA)[
+          shoppingTag[0].getAttribute('data-tag-id')
+        ]['product-title'];
+
+        const shopLocalizedString =
+          this.localizationService_.getLocalizedString(
+            LocalizedStringId_Enum.AMP_STORY_SHOPPING_PRODUCT_NAME
+          );
```

Comment:
Move this outside of mutateElement.
Dom maniuplation should be the only thing in mutateElement.